kernel/mod_typing.ml

```diff
@@ -293,45 +290,45 @@ let rec translate_mse_funct env mpo inl mse = function
     sign, NoFunctor alg, reso, cst
   |(mbid, ty) :: params ->
     let mp_id = MPbound mbid in
-    let mtb = translate_modtype env mp_id inl ([],ty) in
+    let mtb, cst = translate_modtype env mp_id inl ([],ty) in
     let env' = add_module_type mp_id mtb env in
-    let sign,alg,reso,cst = translate_mse_funct env' mpo inl mse params in
+    let sign,alg,reso,cst' = translate_mse_funct env' mpo inl mse params in
     let alg' = MoreFunctor (mbid,mtb,alg) in
-    MoreFunctor (mbid, mtb, sign), alg',reso, cst +++ mtb.mod_constraints
+    MoreFunctor (mbid, mtb, sign), alg',reso, Univ.Constraint.union cst cst'
 
 and translate_modtype env mp inl (params,mte) =
   let sign,alg,reso,cst = translate_mse_funct env None inl mte params in
-  let mtb = mk_modtype (mp_from_mexpr mte) sign cst reso in
+  let mtb = mk_modtype (mp_from_mexpr mte) sign reso in
   let mtb' = subst_modtype_and_resolver mtb mp in
-  { mtb' with mod_type_alg = Some alg }
+  { mtb' with mod_type_alg = Some alg }, cst
 
 (** [finalize_module] :
     from an already-translated (or interactive) implementation and
     an (optional) signature entry, produces a final [module_body] *)
 
-let finalize_module env mp (sign,alg,reso,cst) restype = match restype with
-  |None ->
+let finalize_module env mp (sign,alg,reso,cst1) restype = match restype with
```

Comment:
This `cst1` shouldn't be passed as an argument.

Reply:
This is annoying to change with the way the `translation` type and code plumbing work, feel free to push a patch on top but not worth the effort for me.